client/src/context/gameContext.js

```diff
@@ -49,11 +55,41 @@ export const GameProvider = (props) => {
   };
 
   const cardUpdate = (data) => {
-    // console.log('listening');
+    const updatedCards = cardsRef.current;
+
+    const updatedIndex = updatedCards.findIndex(
+      (element) => element.id === data.id
+    );
+
+    if (updatedIndex < 0) {
+      return;
+    }
+
+    /*
+    console.log(`Changing zone of card ${data.id} from `
+    + `${updatedCards[updatedIndex].state.zone} to `
+    + `${data.state.zone}`);
+    */
+
+    updatedCards[updatedIndex].state = data.state;
+
+    dispatch({ type: GameActions.UPDATE_CARD, payload: updatedCards });
   };
 
   const playerUpdate = (data) => {
-    // console.log('listening');
+    const updatedPlayers = playersRef.current;
```

Comment:
Same as above, could add `slice()` here

Reply:
Good point; I'll go ahead and make this and the above changes